nanopub/profile.py

```diff
@@ -20,46 +14,86 @@ class ProfileError(RuntimeError):
     pass
 
 
-def get_orcid_id():
-    orcid_id = get_profile()[ORCID_ID]
-    if not orcid_id:
-        raise ProfileError('Your profile was not setup yet or not setup correctly. To setup '
-                           'your profile see instructions in Readme.md')
-    return orcid_id
+class Profile:
+    """Represents a user profile.
+
+    Attributes:
+        orcid_id: The user's ORCID
+        name: The user's name
+        public_key: Path to the user's public key
+        private_key: Path to the user's private key
+        nanopub_uri: URI of the user's profile nanopub
+    """
+    def __init__(
+            self,
+            orcid_id: str, name: str,
+            public_key: Path, private_key: Path,
+            nanopub_uri: Optional[str] = None
```

Comment:
`Profile.profile_nanopub` sounded silly, so I'm calling it `nanopub_uri` on the Python side.

Reply:
I think we should rename this everywhere while we're at it. Maybe into `introduction_nanopub_uri` to make it even clearer.

Reply:
Well, you can't change the YAML format I think? Unless you want to modify the Java code as well?

Reply:
No, the yaml format is ours! The java code has nothing to do with this. So we can change anything! We could even change `profile_nanopub` to `lourens_is_awesome`.

Reply:
Changed to `introduction_nanopub_uri` now, and removed the mapping code so that the YAML format now matches the Python class exactly.

Reply:
Actually, if it's all the same to you, maybe `orcid_id` should be `orcid`? It's not the id of the id, but just the id itself.

Reply:
Nice work! We had quite some dicussions about that, we settled on orcid_id. ORCID refers to it as ORCID iD, see https://support.orcid.org/hc/en-us/articles/360006897334-What-is-my-ORCID-iD-and-how-should-I-use-it- . But in fact ORCID stands for Open Researcher and Contributor ID so they refer to it themselves as Open Researcher and Contributor ID ID.

Reply:
Ha! Well, when in Rome...